internal/github/publish_github_gist_test.go

```diff
@@ -216,3 +217,49 @@ func (s *GithubPublishGistTestSuite) TestResetOldFiles() {
 	})
 	s.Require().Nil(diags)
 }
+
+func (s *GithubPublishGistTestSuite) TestError() {
+	s.cli.On("Gists").Return(s.gistCli)
+	s.gistCli.On("Create", mock.Anything, mock.Anything).Return(nil, nil, errors.New("github api failed"))
+	ctx := context.Background()
+	titleMeta := plugindata.Map{
+		"provider": plugindata.String("title"),
+		"plugin":   plugindata.String("blackstork/builtin"),
+	}
+
+	diags := s.plugin.Publish(ctx, "github_gist", &plugin.PublishParams{
+		Config: plugintest.NewTestDecoder(s.T(), s.plugin.Publishers["github_gist"].Config).
+			SetAttr("github_token", cty.StringVal("testtoken")).
+			Decode(),
+		Args: plugintest.NewTestDecoder(s.T(), s.plugin.Publishers["github_gist"].Args).
+			SetAttr("description", cty.StringVal("test description")).
+			SetAttr("filename", cty.StringVal("filename.md")).
+			Decode(),
+		Format: plugin.OutputFormatMD,
+		DataContext: plugindata.Map{
+			"document": plugindata.Map{
+				"meta": plugindata.Map{
+					"name": plugindata.String("test_document"),
+				},
+				"content": plugindata.Map{
+					"type": plugindata.String("section"),
+					"children": plugindata.List{
+						plugindata.Map{
+							"type":     plugindata.String("element"),
+							"markdown": plugindata.String("# Header 1"),
+							"meta":     titleMeta,
+						},
+						plugindata.Map{
+							"type":     plugindata.String("element"),
+							"markdown": plugindata.String("Lorem ipsum dolor sit amet, consectetur adipiscing elit."),
+						},
+					},
+				},
+			},
+		},
+		DocumentName: "test_doc",
+	})
+	s.Require().Len(diags, 1)
+	s.Require().Equal("Failed to create gist", diags[0].Summary)
+	s.Require().Equal("github api failed", diags[0].Detail)
```

Comment:
I suggest using `diagtest` to match diagnostics with expected output, and only check 
```go
diagtest.IsError,
diagtest.DetailContains("github api failed"),
```

We used to check for exact diagnostics in tests, but it was very fragile, and adding features (for example an ability to report error locations in plugins) broke tons of tests in small annoying ways.

Checking just specific fields is less fragile, but still, adding some context to `.Detail` rephrasing the `.Summary` would needlessly break the test that should just check that the error text was reported to user in diagnostic _in some form_

P.S.: unrelated, but be careful when adding tests relating to os errors. Found out that windows has completely different error messages for stuff like "file not found" (CI will catch this soon)

Reply:
Thank you